### PR TITLE
Light Boolean search implementation

### PR DESCRIPTION
**Pull request by Alkarex**
`Hello intitle:World OR date:P1D example OR author:Else intitle:"This or that"`
https://github.com/FreshRSS/FreshRSS/issues/879

### CODE REVIEW

**Comment by Alkarex** [comment]
All parts are implicitly `AND` (which should not be written), except if `OR` is stated.
No use of parentheses. Support for quotes to disable the Boolean search, like `"This or that"`.

**Review by Frenzie — APPROVED** [review]
Looks pretty straightforward.

**Comment by Alkarex** [comment]
It was not the case before this patch either, but we so far do not support quotes for free text:
`"Hello World"`.
We do already support quotes in more precise searches like `author:"Jean Dupont"`

File: docs/en/users/03_Main_view.md [hunk 1]
```diff
@@ -181,3 +181,6 @@ Some operators can be used negatively, to exclude articles, with the same syntax
 `-author:name`, `-intitle:keyword`, `-inurl:keyword`, `-#tag`, `!keyword`.
 
 It is also possible to combine operators to have a very sharp filter, and it is allowed to have multiple instances of `author:`, `intitle:`, `inurl:`, `#`, and free-text.
+
+Combining several search criteria implies a logical *and*, but the keywoard ` OR ` can be used to combine several search critetia with a logical *or* instead:
```
Comment (Frenzie) [inline docs/en/users/03_Main_view.md]: critetia :-P

**Comment by Alkarex** [comment]
Thanks, fixed :-) https://github.com/FreshRSS/FreshRSS/pull/1828/commits/4c767fb20f73284d276b5fd476f6c1f453f6826c

**Comment by Alkarex** [comment]
Let's merge to increase the testing

**Pull request merged**